tests/unit/test_charm.py

```diff
@@ -111,27 +120,30 @@ def test_pollen_prepare(
         glob_mock.return_value = None
         path_mock.return_value = False
         self.pollen.prepare("pollen-0", CharmState("hostname"))
-        apt_update_mock.assert_called_once()
         service_restart_mock.assert_called_once_with("rsyslog.service")
 
-    @mock.patch("charms.operator_libs_linux.v0.apt.update")
+    @mock.patch("charms.operator_libs_linux.v2.snap.add")
     def test_pollen_prepare_install_error(
         self,
-        apt_update_mock,
+        snap_mock,
     ):
-        apt_update_mock.side_effect = FileNotFoundError
+        snap_mock.side_effect = snap.SnapError
         with self.assertRaises(exceptions.InstallError):
             self.pollen.prepare("pollen-0", CharmState("hostname"))
 
     @mock.patch("charms.operator_libs_linux.v1.systemd.service_restart")
-    @mock.patch("subprocess.run")
+    @mock.patch("charms.operator_libs_linux.v2.snap.add")
+    @mock.patch("pathlib.Path.read_bytes")
```

Comment:
Perhaps it is better to let the tests interact with the real file system?

Reply:
In the unit test ? Sounds a bit strange especially with the path used in the prepare method

Reply:
Can't the file just be created, e.g., in a temporary directory?

Reply:
Look at the prepare method, how would you do that ?
https://github.com/canonical/pollen-operator/blob/daa6ffe8fc92b211cca15b9f206b92ae11f00d2d/src/pollen.py#L21

Reply:
The paths are fixed, so I don't see how using a temp directory would be possible.